xarray/core/missing.py

```diff
@@ -176,6 +176,15 @@ def get_clean_interp_index(arr, dim, use_coordinate=True, **kwargs):
         # raise if index cannot be cast to a float (e.g. MultiIndex)
         try:
             index = index.values.astype(np.float64)
+            if method != 'nearest':
```

Comment:
rescaling is necessary to avoid underflow/overflow RuntimeWarnings with datetime64. I've made a similar change to interp() below.